pybossa/util.py

```diff
@@ -950,13 +950,17 @@ def valid_or_no_s3_bucket(task_data):
 
 
 def can_update_user_info(current_user, user_to_update):
+    # admin can update anyone
     if current_user.admin:
         return True
-    if current_user.subadmin and current_user.id == user_to_update.id:
-        return True
-    if not current_user.subadmin:
-        return False
-    return not (user_to_update.admin or user_to_update.subadmin)
+    # subadmin can update self and normal users
+    if current_user.subadmin:
+        return (current_user.id == user_to_update.id or
+            not (user_to_update.admin or user_to_update.subadmin))
+    # normal user can update self except for 'user_type' field
+    if current_user.id == user_to_update.id:
+        return {'disabled':{'user_type': 'You must be an admin or subadmin to edit this.'}}
+    return False
```

Comment:
you could return status, status_info from this function; status_info can be empty when no additional info to be passed about user_type disabled.

Reply:
Agree

Reply:
When you say "status info can be empty" do you mean `None`? Or do you mean empty `dict`? Either one will work.

If I do that I have to change every call site, which is expecting a boolean return value, to now get the first element from the returned tuple.

The way I did it doesn't require any changes to the call sites since the return value, when converted to bool, gives the correct result for can_update.